src/backend/distributed/planner/recursive_planning.c

```diff
@@ -528,7 +530,7 @@ RecursivelyPlanSubquery(Query *subquery, RecursivePlanningContext *planningConte
 	subPlan = CreateDistributedSubPlan(subPlanId, subquery);
 	planningContext->subPlanList = lappend(planningContext->subPlanList, subPlan);
 
-	resultQuery = BuildSubPlanResultQuery(subquery, planId, subPlanId);
+	resultQuery = BuildSubPlanResultQuery(subquery, NIL, planId, subPlanId);
```

Comment:
Maybe add a single line comment explaining why we're always passing `NIL` such as 
`/*  BuildSubPlanResultQuery() preserves the column aliases (if exists) for subqueries, thus no need to provide them for the result query */`

Reply:
done